packages/flutter/test/foundation/service_extensions_test.dart

```diff
@@ -0,0 +1,296 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:test/test.dart';
+
+class TestServiceExtensionsBinding extends BindingBase
+  with SchedulerBinding,
+       ServicesBinding,
+       GestureBinding,
+       RendererBinding,
+       WidgetsBinding {
+
+  final Map<String, ServiceExtensionCallback> extensions = <String, ServiceExtensionCallback>{};
+
+  @override
+  void registerServiceExtension({
+    @required String name,
+    @required ServiceExtensionCallback callback
+  }) {
+    expect(extensions.containsKey(name), isFalse);
+    extensions[name] = callback;
+  }
+
+  Future<Map<String, String>> testExtension(String name, Map<String, String> arguments) {
+    expect(extensions.containsKey(name), isTrue);
+    return extensions[name](arguments);
+  }
+
+  int reassembled = 0;
```

Comment:
I can if you want but I was grouping it with reassembleApplication since it's closely related (i did the same with the frameScheduled/scheduleFrame/doFrame set)